assembl/static2/js/app/components/debate/common/ideasLevel.jsx

```diff
@@ -59,6 +53,39 @@ class IdeasLevel extends React.Component {
       }, 500);
     }
   }
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.updateDimensions);
+  }
+  setDimensions() {
+    if (window.innerWidth > APP_CONTAINER_MAX_WIDTH) {
+      this.setState({
+        sliderContainerWidth: APP_CONTAINER_MAX_WIDTH + this.getRightOverflowValue(),
+        ideaPreviewWidth: APP_CONTAINER_MAX_WIDTH / NB_IDEA_PREVIEW_TO_SHOW
+      });
+    } else {
+      const ideaPreviewWidth = (window.innerWidth - APP_CONTAINER_PADDING) / (NB_IDEA_PREVIEW_TO_SHOW + 0.5);
+      if (ideaPreviewWidth <= IDEA_PREVIEW_MIN_WIDTH) {
+        this.setState({
+          sliderContainerWidth: window.innerWidth - APP_CONTAINER_PADDING * 2,
+          ideaPreviewWidth: IDEA_PREVIEW_MIN_WIDTH
+        });
+      } else {
+        this.setState({
+          sliderContainerWidth: window.innerWidth - APP_CONTAINER_PADDING * 2,
+          ideaPreviewWidth: ideaPreviewWidth
+        });
+      }
+    }
+  }
+  updateDimensions() {
+    this.setState(
+      {
+        sliderCount: 0,
+        sliderLeftPosition: 0
+      },
+      this.setDimensions()
```

Comment:
Why you callback in `setState`?

Reply:
Because `setState` is asynchronous. I want to call `setDimensions()` immediately after setting state.

Reply:
Euh @paolina92 you shouldn't call it here, it should be just `this.setDimensions` here.